### ISSUE TITLE
Homebridge crashes if radio is unreachable

### ISSUE THREAD

**Issue by Karrypto** [boikedamhuis/homebridge-frontier-silicon#4]
If the wifi radio is unreachable (e.g. bad wireless connection) home bridge completely crashes with an exception..unfortunately, you have to restart homebridge every time...

is there a way to catch this behavior ?

**Comment by Smexhy** [comment]
This is true, but for me all I need is to connect the radio to the internet again, and homebridhe will start working again without restart. But it definitely needs to be patched asap.

**Comment by Karrypto** [comment]
> This is true, but for me all I need is to connect the radio to the internet again, and homebridhe will start working again without restart. But it definitely needs to be patched asap.

**Comment by Karrypto** [comment]
any updates ?

**Comment by ethempekin** [comment]
Great to find this valuable resource plugin but is it working?

**Comment by boikedamhuis** [comment]
This issue is resolved in the current release ✅

The latest version of the plugin now handles unreachable radios gracefully. If the device is temporarily offline or does not respond, the plugin no longer crashes Homebridge — it safely returns the last known state and continues polling without unhandled promise rejections.

Please update to the most recent version, restart Homebridge, and let me know if you still see any instability with offline devices.

Thanks for reporting this!
